tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -21,93 +21,192 @@
 from s3_common_utils import check_s3_read_resource, check_s3_read_write_resource, get_policy_resources
 
 from tests.common.assertions import assert_no_errors_in_logs
+from tests.common.schedulers_common import get_scheduler_commands
+from tests.schedulers.test_awsbatch import _test_job_submission as test_job_submission_awsbatch
+from tests.schedulers.test_slurm import _wait_for_computefleet_changed as wait_for_computefleet_changed
 
 
 @pytest.mark.usefixtures("os", "instance")
 def test_iam_roles(
```

Comment:
optional: can we extract parts of this function in separate self contained local methods?it's getting very complex and hard to read. For example we could have `_test_cluster_create` `_test_cluster_update` `_test_cluster_scaling`